types/src/transaction/authenticator.rs

```diff
@@ -461,6 +470,13 @@ impl AuthenticationKeyPreimage {
         Self::new(public_key.to_bytes(), Scheme::MultiEd25519)
     }
 
+    pub fn uuid(transaction_derived_uuid: TransactionDerivedUUID) -> AuthenticationKeyPreimage {
+        let mut hash_arg = Vec::new();
+        hash_arg.push(Scheme::DeriveUuid as u8);
+        hash_arg.extend(transaction_derived_uuid.hash().to_vec());
```

Comment:
Here, you do not want to hash `hash_arg`. Just return `Self(hash_arg)`. The hashing will happen by calling `AuthenticationKey::from_preimage()` over this returned `AuthenticationKeyPreimage` in the right place. (i.e., in your native).

Reply:
@alinush  - should we be doing two hashes ? (i.e. transaction_derived_uuid.hash() and then AuthenticationKey::from_preimage on top)

should we instead of transaction_derived_uuid.hash()  use bcs::to_bytes(transaction_derived_uuid) ? 

are all other places that use AuthenticationKeyPreimage doing double hashing?

Reply:
@alinush Done.

Reply:
Igor, I think so. We might've overdone it here a little, since I didn't originally understand this PR.

Be sure to remove the `CryptoHash` traits from the struct too.

Reply:
I changed the implementation to use only one hash.